File: backend/app/models/district.py
# ─── models/district.py ───────────────────────────────────────────────────────
from pydantic import BaseModel, Field
from typing import Optional

class District(BaseModel):
    dist_name:          str
    state_name:         str
    latitude:           float
    longitude:          float
    population:         int
    dfsi_score:         float          # From DFSI.csv
    pct_flooded_area:   float          # From District_FloodedArea.csv
    human_fatality:     int            # From District_FloodImpact.csv
    human_injured:      int
    mean_flood_duration:float
    road_density:       Optional[float] = None   # From catchment_characteristics
    elevation:          Optional[float] = None
    ruggedness:         Optional[float] = None
    demand_per_period:  Optional[float] = None   # Computed by demand_estimator
    uav_cost:           Optional[float] = None   # Computed by cost_calculator
    truck_cost:         Optional[float] = None

class DistrictInDB(District):
    id: Optional[str] = Field(None, alias="_id")

File: express-server/src/models/District.js
// ─── src/models/District.js ───────────────────────────────────────────────────
const { Schema, model } = require("mongoose");
const DistSchema = new Schema({
  dist_name:    String,
  state_name:   String,
  latitude:     Number,
  longitude:    Number,
  dfsi_score:   Number,
  demand_mean:  Number,
  truck_cost:   Number,
  uav_cost:     Number,
  population:   Number,
});
module.exports = model("District", DistSchema);
